api/v1alpha1/localbuild_types.go

```diff
@@ -45,8 +45,19 @@ type PackageConfigsSpec struct {
 	CorePackageCustomization map[string]PackageCustomization `json:"packageCustomization,omitempty"`
 }
 
+// BuildCustomizationSpec fields cannot change once a cluster is created
+type BuildCustomizationSpec struct {
+	Protocol       string `json:"protocol,omitempty"`
+	Host           string `json:"host,omitempty"`
+	IngressHost    string `json:"ingressHost,omitempty"`
+	Port           string `json:"port,omitempty"`
+	UsePathRouting bool   `json:"usePathRouting,omitempty"`
```

Comment:
I mean maybe a nit but do we ever expect to have other kind of routing where we want to make this an enum string?

Reply:
Hmm I honestly cannot think of other ways to route for now. Maybe this is something we can take a look at later along with the potential changes to the name of the flag. #417